hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/RoutineEnvironmentTransformations.scala

```diff
@@ -0,0 +1,414 @@
+    val daytimeSleepLocationOtherDescription =
+      if (daytimeSleepLocation.contains(98L)) {
+        rawRecord.getOptional("de_sleep_loc_day_other")
+      } else None
+    dog.copy(
+      deNighttimeSleepLocation = nighttimeSleepLocation,
+      // ONLY if: de_sleep_location = '98'
+      deNighttimeSleepLocationOtherDescription = if (nighttimeSleepLocation.contains(98L)) {
+        rawRecord.getOptional("de_sleep_location_other")
+      } else None,
+      deNighttimeSleepAvgHours = rawRecord.getOptionalNumber("de_sleep_amt_night"),
+      deDaytimeSleepLocationDifferent = daySleep,
+      deDaytimeSleepLocation = daytimeSleepLocation,
+      deDaytimeSleepLocationOtherDescription = daytimeSleepLocationOtherDescription,
+      deDaytimeSleepAvgHours = rawRecord.getOptionalNumber("de_sleep_amt_day")
+    )
+  }
+
+  /* Toxins ingested */
+  def mapToxinsIngested(
+    rawRecord: RawRecord,
+    dog: HlesDogRoutineEnvironment
+  ): HlesDogRoutineEnvironment = {
+    val toxinsAmount = rawRecord.getOptionalNumber("de_ingest_bad_amt")
+    //ONLY if: de_ingest_bad_amt = 1 OR de_ingest_bad_amt = 2
+    if (toxinsAmount.exists(_ > 0)) {
+      //val toxinsDescription = rawRecord.getOptional("de_ingest_bad") //FIXME REMOVE
+      val toxinsDescription = rawRecord.getOptional("de_ingest_bad").map(_.trim).filter(_.nonEmpty)
+      val toxinsIngested = rawRecord.get("de_ingest_bad_what")
+      val recentToxinsOrHazardsIngestedOther = toxinsIngested.map(_.contains("98"))
+      dog.copy(
+        deRecentToxinsOrHazardsIngestedFrequency = toxinsAmount,
+        deRecentToxinsOrHazardsIngestedChocolate = toxinsIngested.map(_.contains("1")),
+        deRecentToxinsOrHazardsIngestedPoison = toxinsIngested.map(_.contains("2")),
+        deRecentToxinsOrHazardsIngestedHumanMedication = toxinsIngested.map(_.contains("3")),
+        deRecentToxinsOrHazardsIngestedPetMedication = toxinsIngested.map(_.contains("4")),
+        deRecentToxinsOrHazardsIngestedGarbageOrFood = toxinsIngested.map(_.contains("5")),
+        deRecentToxinsOrHazardsIngestedDeadAnimal = toxinsIngested.map(_.contains("6")),
+        deRecentToxinsOrHazardsIngestedToys = toxinsIngested.map(_.contains("7")),
+        deRecentToxinsOrHazardsIngestedClothing = toxinsIngested.map(_.contains("8")),
+        deRecentToxinsOrHazardsIngestedOther = recentToxinsOrHazardsIngestedOther,
+        //de_ingest_bad is deprecated, check it first, then check de_ingest_bad_what_other
+        deRecentToxinsOrHazardsIngestedOtherDescription =
+          //  if (toxinsDescription.exists(_.trim.nonEmpty)) {
+          //    toxinsDescription
+          //    // ONLY if: de_ingest_bad_what(98) = 1
+          //  } else if (recentToxinsOrHazardsIngestedOther.get) {
+          //    rawRecord.getOptional("de_ingest_bad_what_other")
+          //  } else None,
+          // ONLY if: de_ingest_bad_what(98) = 1
+          recentToxinsOrHazardsIngestedOther.fold(toxinsDescription) {
```

Comment:
I implemented those changes but it's failing one of my unit test cases where:
"de_ingest_bad" (toxinsDescription) is provided and "de_ingest_bad_what_other" is not defined.
I get the error: 
The Option on which value was invoked was not defined. (RoutineEnvironmentTransformationsSpec.scala:559)

In that case,
"de_ingest_bad" -> Array("Some nasty stuff"),
"de_ingest_bad_what" -> Array("1", "2", "3", "4", "5"), (OTHER option not indicated)

If I am understanding the logic correctly, we should be checking toxinsDescription (de_ingest_bad) first and if that isn't available, pull "de_ingest_bad_what_other" (but that field only exists when de_ingest_bad_what(98) = 1).

Reply:
Ah, I see the issue. Your test examples include both "de_ingest_bad" and "de_ingest_bad_what", but this code is written to assume that only one or the other will ever be present. I believe the assumption is correct: when they introduced de_ingest_bad_what, they removed de_ingest_bad, so we should never see the two in a single record.

Reply:
That makes sense! I made sure all the test data had either the deprecated field OR the new one. I just had to make one change, making toxinsDescription the default (instead of None) within that function and now it passes all the unit tests